ocaml/typing/typecore.ml

```diff
@@ -867,6 +867,62 @@ let apply_mode_annots ~loc ~env (m : Alloc.Const.Option.t) mode =
     | Error (s, e) -> error (s, `Linearity e)
     ) m.linearity
 
+(** Takes [m0] which is the parameter on mutable, and [m] which is a left mode
+    on the record being accessed, returns the left mode of the projected mutable
+    field. *)
+let project_mutable m0 m =
```

Comment:
Once the monadic case is removed, this function is no longer needed.

Reply:
I think it is clearer to get rid of them so that it is obvious that projections do not care about the mutability

Reply:
After we introduce syncness/contention, `project_mutable` is good place to check that `m` is uncontended.